check.ml

```diff
@@ -99,19 +79,15 @@ let well_formed prog =
         | Simple e -> check_simple_expr e
         | Op (_op, xs) ->
           List.iter check_simple_expr xs in
-      let check_arg = function
-        | Arg_by_val e -> check_expr e
-        | Arg_by_ref x -> () in
+      let check_arg = check_expr in
       let check_osr = function
```

Comment:
Currently that would be correct. I am thinking of implementing `array x[e]` and `array x[] = [...]` in osr-map, so that we would have several cases again.